### PR TITLE
Feature : Added the feature of sending the notification of latest video/note to discord

### PR DESCRIPTION
**Pull request by amanbairagi30**
### **Problem Statement**
Currently, whenever a new video is uploaded into the CMS, Harkirat has to update the Discord community about it. This process requires additional effort after posting the video in the CMS.

### **Solution**
I have implemented a solution where whenever a new video or note is added from the CMS admin panel, a notification for that video or note can be seen in Discord. I have made this an optional choice. If moderators want to add a video/note without sending a notification to discord, they can do so. Additionally , I have provided a checkbox for this purpose . Along with that in discord message you can even go to that particular video /note location by clicking the link provided there .

You can see the changes below :

https://github.com/code100x/cms/assets/118182376/e8655f37-9d6e-434f-8502-e4f42ce65e8a

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I assure there is no similar/duplicate pull request regarding same issue

### CODE REVIEW

**Comment by amanbairagi30** [comment]
Hey @hkirat , I have made the necessary changes you asked in live stream , and creating a new PR as the old one has some weird merge conflicts . I would request you to review this PR .

**Comment by siinghd** [comment]
@amanbairagi30 

can you add instruction and update .env.example on what do we need to setup the discord webhook url?

**Comment by amanbairagi30** [comment]
> @amanbairagi30
> 
> can you add instruction and update .env.example on what do we need to setup the discord webhook url?
> 
> Also solve the conflicts

Done ⚡

File: src/app/services/DiscordService.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        {
+          title: `New ${data?.type === 'notion' ? 'NOTE' : data?.type?.toUpperCase()}`,
+          description: `${data?.title} has been added in the ${data?.courseTitle} , [Click here to visit this ${data?.type === 'notion' ? 'note' : data?.type}](https://app.100xdevs.com/courses/${data.courseId}/${data.currFolderId}/${data.mediaId})`,
+        },
+      ],
+    };
+
+    try {
+      const res = await axios.post(
+        `${process.env.NEXT_PUBLIC_DISCORD_WEBHOOK_URL}`,
```
Comment (siinghd) [inline src/app/services/DiscordService.tsx:29]: This is being leaked on client site please move this login on server side

File: src/components/admin/AddContent.tsx [hunk 2]
```diff
@@ -2,43 +2,136 @@
...
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+    setLoading(false);
+
+    const respData: { id: number } = await response.json();
+
+    const data = {
```
Comment (siinghd) [inline src/components/admin/AddContent.tsx:67]: This is unnecessary here, do it server side here  at /api/admin/content

**Review by siinghd — CHANGES_REQUESTED** [review]
Please review the changes the requested and also can you add that if there is no webhook in the .env do not execute the request

File: src/components/admin/AddContent.tsx (hunk 2 above)
Comment (amanbairagi30) [reply src/components/admin/AddContent.tsx:67]: so @siinghd , do you want me to shift all of this logic to api/admin/content route and send checked variable in this request ?
````
const response = await fetch('/api/admin/content', {
      body: JSON.stringify({
        type,
        description: '',
        thumbnail: imageUri,
        title,
        courseId,
        parentContentId,
        metadata,
        adminPassword,
      }),
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
    });```

Comment (siinghd) [reply src/components/admin/AddContent.tsx:67]: Yes sir, also if no webhook env is set, do not send the dicord request

Comment (amanbairagi30) [reply src/components/admin/AddContent.tsx:67]: Okay ser , understood . 🙌

Comment (amanbairagi30) [reply src/components/admin/AddContent.tsx]: done ser ✌

File: src/app/services/DiscordService.tsx (hunk 1 above)
Comment (amanbairagi30) [reply src/app/services/DiscordService.tsx]: done ser ✌

**Comment by amanbairagi30** [comment]
@siinghd , you may check now .

**Comment by amanbairagi30** [comment]
@siinghd sir ji , please look into this .

**Comment by amanbairagi30** [comment]
@siinghd ?

**Comment by amanbairagi30** [comment]
closing it as it has been merged to main via another PR

**Pull request closed without merging**